docs/proposals/storage-capacity-proposal.md

```diff
@@ -0,0 +1,132 @@
+    verbs: ["get", "list", "watch", "create", "update", "patch", "delete"]
+  # The GET permissions below are needed for walking up the ownership chain
+  # for CSIStorageCapacity. They are sufficient for deployment via
+  # StatefulSet (only needs to get Pod) and Deployment (needs to get
+  # Pod and then ReplicaSet to find the Deployment).
+  - apiGroups: [""]
+    resources: ["pods"]
+    verbs: ["get"]
+  - apiGroups: ["apps"]
+    resources: ["replicasets"]
+    verbs: ["get"]
+```
+To enable using these object during scheduling `storageCapacity: true` must be added to CSIDriver spec.
+
+To implement `GetCapacity` it is possible to use AC. `GetCapacity` types look like this
+
+```
+
+type GetCapacityRequest struct {
+  // These are the same
+  // `volume_capabilities` the CO will use in `CreateVolumeRequest`.
+  VolumeCapabilities []*VolumeCapability
+  // These are the same `parameters` the CO will
+  // use in `CreateVolumeRequest`.
+  Parameters map[string]string
+  // This is the same as the
+  // `accessible_topology` the CO returns in a `CreateVolumeResponse`.
+  AccessibleTopology   *Topology
+}
+
+type GetCapacityResponse struct {
+  // The available capacity, in bytes, of the storage that can be used
+  AvailableCapacity int64
+  // The largest size that may be used in a
+  // CreateVolumeRequest.capacity_range.required_bytes field
+  // to create a volume with the same parameters as those in
+  // GetCapacityRequest.
+  MaximumVolumeSize *wrappers.Int64Value
+  // The smallest size that may be used in a
+  // CreateVolumeRequest.capacity_range.limit_bytes field
+  // to create a volume with the same parameters as those in
+  // GetCapacityRequest.
+  MinimumVolumeSize    *wrappers.Int64Value
+}
+```
+
+For each segment and each storage class, CSI GetCapacity is called once with the topology of the segment and the parameters of the class. For each request we should list all
+AC in cluster, filter out AC that do not satisfy segment (node), parameters and volume capacilities of request and find total size of remaining AC for `AvailableCapacity`, max AC
+for `MaximumVolumeSize` and, in case of non-LVG SC, min AC for `MinimumVolumeSize` or minimum size of logical disk (10 Mi) for LVG SC.
+
+Removing ACR require changes in `CreateVolume` function (and others). Storage Capacity helps scheduler
+to pick node for pod, but drive selection algorithm must be implemented by developers and may depend on AC as well. 
+
+## Assumptions (if applicable)
+
+| ID      |     Name     | Descriptions | Comments |
+|---------|--------------|--------------|----------|
+| ASSUM-1 |  Performance | Scheduling time may decrease             |  Decreasing of scheduling time was noted in test Kind cluster with disabled Extender |
+
+## Open issues (if applicable)
+
+| ID      | Name | Descriptions | Status | Comments |
+|---------|------|--------------|--------|----------|
+| ISSUE-1 |      |              |        |          |   
```

Comment:
https://kubernetes.io/docs/concepts/storage/storage-capacity/#limitations 

> One situation where scheduling can fail permanently is when a Pod uses multiple volumes: one volume might have been created already in a topology segment which then does not have enough capacity left for another volume. Manual intervention is necessary to recover from this, for example by increasing capacity or deleting the volume that was already created.

This must be mentioned as open issue

Reply:
Added issue